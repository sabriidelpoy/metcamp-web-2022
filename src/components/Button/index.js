import './styles.css';

function Button({ text, onClick, classes }) {
    return (
        <button onClick={onClick} className={`button ${classes}`}>
            { text }
        </button>
    )
}

export default Button;
