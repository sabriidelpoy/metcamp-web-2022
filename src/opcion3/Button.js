import React from "react";

export function Button({ text, onClickHandler }) {
    return (
        <button onClick={onClickHandler} className="button is-primary">
            { text }
        </button>
    )
}