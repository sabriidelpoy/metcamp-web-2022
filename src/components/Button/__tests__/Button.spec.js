import { render, screen, fireEvent } from "@testing-library/react";

import Button from '../index';

const props = {
    text: 'MetCamp Quiz test',
    classes: 'is-primary is-outlined is-large',
    onClick: jest.fn(),
}

describe('Button Component', () => {
    test('should match snapshot', () => {
        const { container } = render(<Button {...props} />);
        expect(container).toMatchSnapshot();
    });

    test('should ejecute onClick function', () => {
        // 1. Inicialización
        render(<Button {...props} />);
        // 2. Estímulo
        const buttonHome = screen.getByRole('button', {
            name: /metcamp quiz test/i
          })
        // const buttonHome = screen.getByText('MetCamp Quiz test');
        fireEvent.click(buttonHome);
        // 3. Observar el comportamiento esperado
        expect(props.onClick).toHaveBeenCalled();
    });

});