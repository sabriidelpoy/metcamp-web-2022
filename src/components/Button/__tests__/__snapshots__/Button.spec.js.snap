// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should match snapshot 1`] = `
<div>
  <button
    class="button is-primary is-outlined is-large"
  >
    MetCamp Quiz test
  </button>
</div>
`;
